File: solutions/leetcode/contests/w-438/1-check_if_digits_are_equal_in_strings_after_operations-i.scala
object Solution {
    def hasSameDigits(s: String): Boolean = {
        var str = s
        while (str.length > 2) {
            str = str.sliding(2).map(pair => ((pair(0) - '0' + pair(1) - '0') % 10).toString).mkString
        }
        str(0) == str(1)
    }
}

File: solutions/leetcode/contests/w-438/1-check_if_digits_are_equal_in_strings_after_operations-i.java
class Solution {
    public boolean hasSameDigits(String s) {
        while (s.length() > 2) {
            StringBuilder newString = new StringBuilder();
            for (int i = 0; i < s.length() - 1; i++) {
                int sum = (s.charAt(i) - '0' + s.charAt(i + 1) - '0') % 10;
                newString.append(sum);
            }
            s = newString.toString();
        }
        return s.charAt(0) == s.charAt(1);
    }
}
